### PR TITLE
README badge links, lints, and clippy

### PR DESCRIPTION
**Pull request by shnewto**
cool project! this PR is just a pile of little fiddly things

- some of the readme badges just took you to the badge svg when you clicked them, now they should take you somewhere useful for the stat / staus
- markdownlint had a lot of things to say about the readme so addressed those
- clippy had opionions about initializing structs in place vs creating then mutating them

feel free to pick and choose what things you like and drop the things you dont 😄

### CODE REVIEW

**Comment by davepmiller** [comment]
> * some of the readme badges just took you to the badge svg when you clicked them, now they should take you 
somewhere useful for the stat / staus

🤯

> * markdownlint had a lot of things to say about the readme so addressed those

What tool are you using for markdownlint? I like the idea of maybe bringing that into the CI action

> * clippy had opionions about initializing structs in place vs creating then mutating them

This rules! I was wondering about a better way to handle this.

**Review by davepmiller — COMMENTED** [review]
I'm not sure why the format step is failing... wonder if it has something to do with the fork?

**Comment by davepmiller** [comment]
Oh man, @shnewto. I just also discovered the `--pedantic` flag in `clippy`.
I might have to just add these fixes and push directly! Maybe I'll take your markdownlint changes? Messing with it now.

**Review by davepmiller — APPROVED** [review]

**Comment by shnewto** [comment]
sounds good to me! weird that the doc step is failing though, thought i ran 'cargo doc' locally to verify it was okay but might've changed things between that and pushing 🤔

**Comment by shnewto** [comment]
wrt the markdown stuff, it's a vscode plugin for 'markdownlint'. i can try and dig up some deets but it's mostly stuff that's consistent across md linters, like space before and after titles, and space before and after code blocks.

**Comment by shnewto** [comment]
took a pass at resolving the merge conflicts, might be worth ya taking another pass to make sure things still look good to ya

File: README.md [hunk 1]
```diff
@@ -104,9 +104,9 @@
 * Add parameters to `zone create` for customization
 * Homebrew setup
 
-## Feature Requests
+### Feature Requests?
```
Comment (davepmiller) [inline README.md]: I think the markdown linter will cough on this one. It doesn't like the extra '#'.
And the '?' at the end of a title string.

File: src/dns.rs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (davepmiller) [inline src/dns.rs]: Is the formatter upset about this one? It's waiting for me to approve the CI runs, I'll change that.

**Comment by davepmiller** [comment]
> wrt the markdown stuff, it's a vscode plugin for 'markdownlint'. i can try and dig up some deets but it's mostly stuff that's consistent across md linters, like space before and after titles, and space before and after code blocks.

I missed this! Yeah I ended up going with one that had a github action. [markdown-lint](https://github.com/avto-dev/markdown-lint)
Here's what I've been doing locally:

```bash
docker run --rm \
    -v "$(pwd)/README.md:/README.md:ro" \
    avtodev/markdown-lint:v1 \
    /README.md 
```

It has some nice CHANGELOG-specific features as well:
```bash
docker run --rm \
    -v "$(pwd)/CHANGELOG.md:/CHANGELOG.md:ro" \
    avtodev/markdown-lint:v1 \
    --rules /lint/rules/changelog.js \
    --config /lint/config/changelog.yml \
    /CHANGELOG.md
```

**Pull request merged**